packages/updating-element/src/updating-element.ts

```diff
@@ -926,3 +1021,55 @@ export abstract class UpdatingElement extends HTMLElement {
 // Apply polyfills if available
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 (globalThis as any)['updatingElementPlatformSupport']?.({UpdatingElement});
+
+// Dev mode warnings...
+if (DEV_MODE) {
+  const requestUpdateThenable = {
+    then: (
+      onfulfilled?: (value: boolean) => void,
+      _onrejected?: () => void
+    ) => {
+      console.warn(
+        `\`requestUpdate\` no longer returns a Promise.` +
+          `Use \`updateComplete\` instead.`
+      );
+      if (onfulfilled !== undefined) {
+        onfulfilled(false);
+      }
+    },
+  };
+
+  const requestUpdate = UpdatingElement.prototype.requestUpdate;
+  UpdatingElement.prototype.requestUpdate = function (
+    this: UpdatingElement,
+    name?: PropertyKey,
+    oldValue?: unknown,
+    options?: PropertyDeclaration
+  ) {
+    requestUpdate.call(this, name, oldValue, options);
+    return requestUpdateThenable;
+  };
+}
+
+if (DEV_MODE) {
+  // Default warning set.
+  UpdatingElement.enabledWarnings = ['change-in-update'];
+  const ensureOwnWarnings = function (ctor: typeof UpdatingElement) {
+    if (!ctor.hasOwnProperty('enabledWarnings')) {
```

Comment:
I don't think we need the rest of this code. Users can just write:

```js
UpdatingElement.enabledWarnings?.push('my-warning')
```

This will work in prod mode, as the push() is skipped. And duplicates will be fine, as the code that checks enabled warnings will get a hit on the first duplicate. There probably won't be many duplicates since this is debug code, and perf won't be a problem with small lists.

But also my comment about using an array was assuming that users could just assign the whole array:

```js
UpdatingElement.enabledWarnings = ['my-warning']
```

If that's not viable because you don't want them to override the default of `['change-in-update']`, then the Set is just fine too, since `UpdatingElement.enabledWarnings?.push('my-warning')` isn't so different from `UpdatingElement.enabledWarnings?.add('my-warning')`.

Reply:
The array is more ergonomic for setting the warnings as a block (which seems reasonable to support), but the point of the API `enableWarning/disableWarning` is to be able to customize warnings per class, copying down your superclass' settings. The uniqueness check in the array is so that you can disable a warning (can't have duplicates for that).

This may be fancier than we need, but since we're going to the trouble of having dev mode and it's basically "free code," it seemed worth it.

Reply:
As noted above, we're always carrying the `enableWarning` and `disableWarning` APIs around, even in prod.  I don't think we need a fancy API here, especially for disabling warnings. With an optional array devs can either add to the list, replace the list, remove the whole list, or do fancy things if they care, but I doubt it'll be necessary.

The use cases I see:

- Add a warning:

```ts
UpdatingElement.enabledWarnings?.push('my-warning');
```

- Replace all warnings:

```ts
UpdatingElement.enabledWarnings = ['my-warning'];
```

- Remove all warnings:

```ts
UpdatingElement.enabledWarnings = undefined;
```


- Remove one warning (will this ever be needed?):

```ts
UpdatingElement.enabledWarnings = UpdatingElement.enabledWarnings.filter((w) => w !== 'my-warning');
```